frontend/src/metabase/query_builder/components/NativeQueryEditor/NativeQueryEditor.tsx

```diff
@@ -909,6 +921,14 @@ const mapDispatchToProps = (dispatch: Dispatch) => ({
     );
     return Questions.HACK_getObjectFromAction(action);
   },
+  onSetDatabaseId: (id: DatabaseId) => {
```

Comment:
Please remove `shouldPersistLastUsedDatabase`, add a callback to `NativeQueryEditor` like `onSetDatabaseId` and put the action in `View` and not here

Reply:
Help me understand this better.
What are we trying to achieve by moving the dispatch up to the `View`?

The reason I chose this approach was because I wanted a fine-grained control of when exactly do we persist the database. NativeQueryEditor can be rendered by many different components.

How are we going to control when does native query editor persists the database ID, and when does it ignore it?

Reply:
To answer my own question - this should do it.
https://github.com/metabase/metabase/pull/38986/commits/4dd6aad70815b5780baf5a675b2f62ed98146ebc

I'd still like to learn why is one approach considered better than the other?